Fill.py

```diff
@@ -907,22 +893,12 @@ def failed(warning: str, force: typing.Union[bool, str]) -> None:
                 items = item_list
             if isinstance(items, str):
                 items = [items]
-            block['items'] = items
+            new_block["items"] = items
 
-            locations: block_value = []
-            if 'location' in block:
-                locations = block['location']  # just allow 'location' to keep old yamls compatible
-            elif 'locations' in block:
-                locations = block['locations']
+            locations: block_value = block.locations
```

Comment:
Update: Oh, they were technically supported before this branch, though the weights don't work. Now they throw hashability errors

Reply:
Update again: Or not... well something's broken for sure, guess it's time to start digging

Reply:
So previously, this completely ridiculous code existed https://github.com/ArchipelagoMW/Archipelago/blob/main/Fill.py#L920-L924
This never did anything because of `new_block["locations"] = list(dict.fromkeys(locations))`. Your branch completely removes that code, but it also still allows `locations` to be a dictionary, which always throws hashability errors

Reply:
I think a pretty good way to do this is to do groups first, but instead of adding directly to `locations` you add them to something like `expanded_groups` and then add them all at once